pyproject.toml

```diff
@@ -27,55 +27,44 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
   "Typing :: Typed",
 ]
-dynamic = ["version"]
-dependencies = [
-    "loguru >= 0.7.0",
-]
 
-[project.optional-dependencies]
-# Use `constraints.txt`
-tests = [
-  "coverage",
-  "pytest",
-  "pytest-cov",
-]
-docs = [
-  "furo",
-  "myst-parser",
-  "Sphinx",
-  "sphinxcontrib-spelling",
-]
-stubs = [
-  # Typing stubs go here
-]
-typing = [
-  "crustypy-instance[stubs]",
-  "mypy",
-  "mypy-extensions",
-  "nox",
-  "pytest",
-]
-dev = [
-  "crustypy-instance[tests,docs,stubs,typing]",
-  "black",
-  "build",
-  "nox",
-  "pre-commit",
-  "python-semantic-release",
-  "ruff",
-  "semver",
-  "toml",
-]
+[tool.poetry.dependencies]
+python = "^3.8"
+loguru = "^0.7.0"
+
+[tool.poetry.group.tests.dependencies]
+coverage = "^7.2.7"
+pytest = "^7.4.0"
+pytest-cov = "^4.1.0"
+
+[tool.poetry.group.docs.dependencies]
+furo = "^2023.5.20"
+myst-parser = "^2.0.0"
+Sphinx = "^7.0.1"
+sphinxcontrib-spelling = "^8.0.0"
+
+[tool.poetry.group.stubs.dependencies]
+# Typing stubs go here
+
+[tool.poetry.group.typing.dependencies]
+# Also requires group `stubs`
+mypy = "^1.4.0"
+mypy-extensions = "^1.0.0"
```

Comment:
are you using this explicitly? or is it just a `mypy` dependency?

Reply:
I'm not exactly sure why I added it. Does Mypy install it by default? I'll see if I can remember why I added.

Reply:
yes, it's part of `mypy` dependencies: https://github.com/python/mypy/blob/310b914b6046830aafb0468f65464d75362991ab/setup.py#L223

Reply:
Fair enough, it is indeed a dependency of Mypy, so I guess it makes no sense for me to install it directly. Thanks for the heads up!

https://github.com/python/mypy/blob/310b914b6046830aafb0468f65464d75362991ab/setup.py#L226